src/ansi-c/ansi_c_declaration.cpp

```diff
@@ -47,8 +49,9 @@ void ansi_c_declaratort::build(irept &src)
     else if(t.id()==ID_merged_type)
     {
       // we always walk down the _last_ member of a merged type
-      assert(!t.subtypes().empty());
-      p=&(t.subtypes().back());
+      auto &merged_type = to_merged_type(t);
+      assert(!merged_type.subtypes().empty());
+      p = &(merged_type.subtypes().back());
```

Comment:
It may be good to add some API wrapping `.subtypes().back()` as, again, that's used in 5 out of 6 use cases.

Reply:
done